pkg/vmtest/integration.go

```diff
@@ -301,3 +245,92 @@ func QEMU(o *Options) (*qemu.Options, error) {
 
 	return &o.QEMUOpts, nil
 }
+
+// ChooseTestInitramfs chooses which initramfs will be used for a given test and
+// places it at the location given by outputFile.
+// Default to the override initramfs if one was given. Else, build an initramfs
+// with the given parameters. If no uinit was provided, the generic one is used.
+func ChooseTestInitramfs(o uroot.Opts, uinit, outputFile string) error {
+	override := os.Getenv("UROOT_INITRAMFS")
+	if len(override) > 0 {
+		log.Printf("Overriding with initramfs %s", override)
+		if err := cp.Copy(override, outputFile); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if len(uinit) == 0 {
+		log.Printf("Defaulting to generic initramfs")
+		uinit = "github.com/u-root/u-root/integration/testcmd/generic/uinit"
+	}
+
+	_, err := CreateTestInitramfs(o, uinit, outputFile)
+	return err
+}
+
+// CreateTestInitramfs creates an initramfs with the given build options and
+// uinit, and writes it to the given output file. If no output file is provided,
+// one will be created, and it is the caller's responsibility to remove it.
```

Comment:
Should "it is the caller's responsibility to remove it" be in a new sentence. Must the caller remove it regardless of whether `outputFile` is set?

Reply:
Hm my assumption was that the user should always be responsible for removing the initramfs file since this function is just for creating it. I just wanted to clarify that that was the case even if they didn't pass in `outputFile` - updated the comment!